dough-items/src/main/java/io/github/bakedlibs/dough/items/CustomItemStack.java

```diff
@@ -1,121 +1,77 @@
-        super(item);
-    }
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.List;
+import java.util.function.Consumer;
 
-    public CustomItemStack(Material type) {
-        super(type);
-    }
+@ParametersAreNonnullByDefault
+public final class CustomItemStack {
 
-    public CustomItemStack(ItemStack item, Consumer<ItemMeta> meta) {
-        super(item);
-        ItemMeta im = getItemMeta();
-        meta.accept(im);
-        setItemMeta(im);
+    private CustomItemStack() {
+        throw new IllegalStateException("Cannot instantiate CustomItemStack");
     }
 
-    public CustomItemStack(Material type, Consumer<ItemMeta> meta) {
-        this(new ItemStack(type), meta);
-
+    public static ItemStack create(ItemStack itemStack, Consumer<ItemMeta> metaConsumer) {
+        return new ItemStackEditor(itemStack).appendMetaConsumer(metaConsumer).create();
     }
 
-    public CustomItemStack(ItemStack item, String name, String... lore) {
-        this(item, im -> {
-            if (name != null) {
-                im.setDisplayName(ChatColor.translateAlternateColorCodes('&', name));
-            }
-
-            if (lore.length > 0) {
-                List<String> lines = new ArrayList<>();
-
-                for (String line : lore) {
-                    lines.add(ChatColor.translateAlternateColorCodes('&', line));
-                }
-                im.setLore(lines);
-            }
-        });
+    public static ItemStack create(Material material, Consumer<ItemMeta> metaConsumer) {
+        return new ItemStackEditor(material).appendMetaConsumer(metaConsumer).create();
     }
 
-    public CustomItemStack(ItemStack item, Color color, String name, String... lore) {
-        this(item, im -> {
-            if (name != null) {
-                im.setDisplayName(ChatColor.translateAlternateColorCodes('&', name));
-            }
-
-            if (lore.length > 0) {
-                List<String> lines = new ArrayList<>();
-
-                for (String line : lore) {
-                    lines.add(ChatColor.translateAlternateColorCodes('&', line));
-                }
-                im.setLore(lines);
-            }
-
-            if (im instanceof LeatherArmorMeta) {
-                ((LeatherArmorMeta) im).setColor(color);
-            }
-            if (im instanceof PotionMeta) {
-                ((PotionMeta) im).setColor(color);
-            }
-        });
+    public static ItemStack create(ItemStack item, @Nullable String name, String... lore) {
+        return new ItemStackEditor(item)
+                .setDisplayName(name)
+                .setLore(lore)
+                .create();
     }
 
-    public CustomItemStack addFlags(ItemFlag... flags) {
-        ItemMeta im = getItemMeta();
-        im.addItemFlags(flags);
-        setItemMeta(im);
-
-        return this;
+    public static ItemStack create(ItemStack item, Color color, @Nullable String name, String... lore) {
```

Comment:
not sure we should have a .create with color tbh, not sure how much it's used - probably better to not but not a huge deal

Reply:
I'm not sure but it's there because there used to be a constructor with that signature. If you're okay with it not existing I'm not opposed to removing it.

Reply:
Yeah I'm not a big fan of it, I'd rather it be more explicitly set.